### ISSUE TITLE
armeria-grpc does not support gRPC Metadata

### ISSUE THREAD

**Issue by dawnbreaks** [line/armeria#1788]
Hi, Now trailers isn't passed to the client, and hope to work in the same way as the official grpc-java. 

```java
new GreeterGrpc.GreeterImplBase() {
      @Override
      public void sayHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
        Metadata trailers = new Metadata();
        trailers.put(DEBUG_INFO_TRAILER_KEY, DEBUG_INFO);
        responseObserver.onError(Status.INTERNAL.withDescription(DEBUG_DESC).asRuntimeException(trailers));
      }
    }
```

**Comment by anuraaga** [comment]
Thanks for the report. This should be fixed by https://github.com/line/armeria/pull/1782. Can you check it to make sure it's the same issue as you're having?

**Comment by anuraaga** [comment]
Oh oops - I didn't look at the code close enough this is for official support for Metadata. I'm currently working on some tricky issues for it but hope we can support it soon :)

**Comment by dawnbreaks** [comment]
Best practice for error handling in gRPC is to include additional error details in the response Metadata. Hope armeria support it soon.

**Comment by anuraaga** [comment]
Filed https://github.com/grpc/grpc-java/issues/5788 to get the conversation going, hope we can make it happen.

Just for reference in case you haven't seen it, Armeria has a flag that can propagate the actual exceptions  themselves. It's not appropriate for all use cases (server/client need to be trusted) but can be even easier than doing manipulation of `Metadata` manually.

http://line.github.io/armeria/server-grpc.html#exception-propagation

**Comment by anuraaga** [comment]
Fixed by #1790
